contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+        @param amount_ Amount of gOHM deposit to remove and return to donor 
+    */
+    function _withdraw(uint256 depositId_, uint256 amount_) internal returns (uint256 amountWithdrawn) {
+        if (isInvalidWithdrawal(amount_)) revert YieldDirector_InvalidWithdrawal();
+
+        DepositInfo storage currDeposit = depositInfo[depositId_];
+        uint256 withdrawableBalance = _toAgnostic(currDeposit.principalAmount);
+
+        if (amount_ >= withdrawableBalance) {
+            amountWithdrawn = _withdrawAllPrincipal(depositId_);
+        } else {
+            _withdrawPrincipal(depositId_, amount_);
+            amountWithdrawn = amount_;
+        }
+
+        emit Withdrawn(msg.sender, recipientLookup[depositId_], amountWithdrawn);
+    }
+
+    /**
+        @notice Redeem available gOHM yield from a specific deposit
+        @param depositId_ Deposit ID to withdraw gOHM yield from
+    */
+    function _redeem(uint256 depositId_) internal returns (uint256 amountRedeemed) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+        if (recipientLookup[depositId_] != msg.sender) revert YieldDirector_NotYourDeposit();
+
+        amountRedeemed = _redeemYield(depositId_);
+        if (amountRedeemed == 0) revert YieldDirector_NoRedeemableBalance();
+
+        uint256[] storage receiptIds = recipientIds[msg.sender];
+        if (depositInfo[depositId_].principalAmount == 0) {
+            _closeDeposit(depositId_);
+            for (uint256 i = 0; i < receiptIds.length; ++i) {
+                if (receiptIds[i] == depositId_) {
+                    // Remove id from depositor's ids array
+                    receiptIds[i] = receiptIds[receiptIds.length - 1]; // Delete integer from array by swapping with last element and calling pop()
+                    receiptIds.pop();
+                    break;
+                }
+            }
+            
+            delete recipientLookup[depositId_];
+        }
+
+        emit Redeemed(msg.sender, amountRedeemed);
+        emit Donated(depositInfo[depositId_].depositor, msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem all available gOHM yield from the vault
+    */
+    function _redeemAll() internal returns (uint256 amountRedeemed) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+
+        amountRedeemed = 0;
+
+        // Have to read the IDs into memory because with each redemption
+        // an ID is removed from the state array
+        uint256[] memory receiptIds = recipientIds[msg.sender];
+        
+        for (uint256 index = 0; index < receiptIds.length; ++index) {
+            uint256 currRedemption = _redeem(receiptIds[index]);
+            amountRedeemed += currRedemption;
+
+            emit Donated(depositInfo[receiptIds[index]].depositor, msg.sender, currRedemption);
+        }
+
+        if (amountRedeemed == 0) revert YieldDirector_NoRedeemableBalance();
```

Comment:
this seems kinda redundant because above you call _redeem and inside redeem() if it redeems nothing then it reverts anyway for one deposit. So this will never be 0.

Reply:
good call